frontend/src/ui/Video.tsx

```diff
@@ -118,36 +79,101 @@ export const Thumbnail: React.FC<ThumbnailProps> = ({
             innerOverlay = t("video.upcoming");
         }
 
-        overlay = <div css={{
-            ...overlayBaseCss,
-            ...currentlyLive ? { backgroundColor: "rgba(200, 0, 0, 0.9)" } : {},
-        }}>
+        const backgroundColor = currentlyLive ? "rgba(200, 0, 0, 0.9)" : undefined;
+
+        overlay = <ThumbnailOverlay {...{ backgroundColor }}>
             {innerOverlay}
-        </div>;
+        </ThumbnailOverlay>;
     } else if (event.syncedData) {
-        overlay = <div css={overlayBaseCss}>{formatDuration(event.syncedData.duration)}</div>;
+        overlay = <ThumbnailOverlay>
+            {formatDuration(event.syncedData.duration)}
+        </ThumbnailOverlay>;
     }
 
-    return (
-        <div css={{
-            position: "relative",
-            transition: "0.2s box-shadow",
-            overflow: "hidden",
-            height: "fit-content",
-            borderRadius: 8,
-            aspectRatio: "16 / 9",
-            ...isDark && {
-                img: {
-                    filter: "brightness(90%)",
-                    transition: "0.1s filter",
-                },
+    return <ThumbnailOverlayContainer {...rest}>
+        {inner}
+        {active && <ActiveIndicator />}
+        {overlay}
+    </ThumbnailOverlayContainer>;
+};
+
+type ThumbnailReplacementProps = {
+    audioOnly: boolean;
+    isDark: boolean;
+    isUpcoming?: boolean;
+}
+export const ThumbnailReplacement: React.FC<ThumbnailReplacementProps> = (
+    { audioOnly, isDark, isUpcoming }
+) => {
+    // We have no thumbnail. If the resolution is `null` as well, we are
+    // dealing with an audio-only event and show an appropriate icon.
+    // Otherwise we use a generic icon.
+    const icon = audioOnly ? <LuVolume2 /> : <LuFilm />;
+
+    return <div css={{
+        display: "flex",
+        height: "100%",
+        alignItems: "center",
+        justifyContent: "center",
+        fontSize: 40,
+        background: "linear-gradient(135deg, #33333380 50%, transparent 0),"
+            + "linear-gradient(-135deg, #33333380 50%, transparent 0)",
+        backgroundSize: "17px 17px",
+        color: "#dbdbdb",
+        backgroundColor: "#292929",
+        ...isDark && {
+            backgroundColor: "#313131",
+            background: isUpcoming
+                ? "linear-gradient(135deg, #48484880 50%, transparent 0),"
+                    + "linear-gradient(-135deg, #48484880 50%, transparent 0)"
+                : "linear-gradient(135deg, #3e3e3e80 50%, transparent 0),"
+                    + "linear-gradient(-135deg, #3e3e3e80 50%, transparent 0)",
+        },
+    }}>{icon}</div>;
+};
+
+type ThumbnailOverlayProps = PropsWithChildren & {
+    backgroundColor?: string;
```

Comment:
I would make that mandatory and remove the default below. There isn't really a sensible "default" IMO. The two used colors are both equally valid, so it feels incorrect to randomly promote one as the "default".